### PR TITLE
[Merged by Bors] - feat(Data/Fintype/Basic): `Finset.univ_map_subtype` lemmas

### PR DESCRIPTION
**Pull request by madvorak**
Co-authored-by: Richard Copley <rcopley@gmail.com>

Discussion:
https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Finset.2Euniv.2Esum.20split

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fintype/Basic.lean [hunk 1]
```diff
@@ -332,6 +332,21 @@ theorem coe_filter_univ (p : α → Prop) [DecidablePred p] : (univ.filter p : S
...
+
+lemma univ_val_map_subtype [Fintype α] (p : α → Prop) [DecidablePred p] [Fintype {a // p a}] :
+    univ.val.map ((↑) : { a // p a } → α) = (univ.filter p).val := by
+  apply (map_val (Function.Embedding.subtype p) univ).symm.trans
+  apply congr_arg
+  apply univ_map_subtype
+
+lemma univ_val_map_subtype_val [Fintype α] (f : α → β)
+    (p : α → Prop) [DecidablePred p] [Fintype {a // p a}] :
+    univ.val.map (fun a : { a // p a } => f a.val) = (univ.filter p).val.map f := by
```
Comment (rosborn) [inline Mathlib/Data/Fintype/Basic.lean:347]:
```suggestion
    univ.val.map (Subtype.restrict p f) = (univ.filter p).val.map f := by
```
I think this is the more canonical spelling?

**Review by rosborn — APPROVED** [review]
These look good to me although I'd suggest renaming `univ_val_map_subtype` to `univ_val_map_subtype_val` and `univ_val_map_subtype_val` to `univ_val_map_subtype_restrict`.

**Comment by madvorak** [comment]
> These look good to me although I'd suggest renaming `univ_val_map_subtype` to `univ_val_map_subtype_val` and `univ_val_map_subtype_val` to `univ_val_map_subtype_restrict`.

Done. Thanks for suggestion!

File: Mathlib/Data/Fintype/Basic.lean (hunk 1 above)
Comment (madvorak) [reply Mathlib/Data/Fintype/Basic.lean]: It is all right; I just had to switch to "full" `rfl`. Thanks for the better spelling!

File: Mathlib/Data/Fintype/Basic.lean [hunk 2]
```diff
@@ -332,6 +332,22 @@ theorem coe_filter_univ (p : α → Prop) [DecidablePred p] : (univ.filter p : S
...
+    univ.val.map ((↑) : { a // p a } → α) = (univ.filter p).val := by
+  apply (map_val (Function.Embedding.subtype p) univ).symm.trans
+  apply congr_arg
+  apply univ_map_subtype
+
+lemma univ_val_map_subtype_restrict [Fintype α] (f : α → β)
+    (p : α → Prop) [DecidablePred p] [Fintype {a // p a}] :
+    univ.val.map (Subtype.restrict p f) = (univ.filter p).val.map f := by
+  rewrite [← univ_val_map_subtype_val, Multiset.map_map, comp_def]
+  rfl
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fintype/Basic.lean:349]:
```suggestion
  rw [← univ_val_map_subtype_val, Multiset.map_map, Subtype.restrict_def]
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**